log/logtest/assertions.go

```diff
@@ -0,0 +1,58 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package logtest // import "go.opentelemetry.io/otel/log/logtest"
+
+import (
+	"slices"
+	"testing"
+
+	"go.opentelemetry.io/otel/log"
+)
+
+// AssertRecordEqual compares two log records, and fails the test if they are
+// not equal.
+func AssertRecordEqual(t testing.TB, want log.Record, got log.Record) {
```

Comment:
Can this return a bool like e.g. https://pkg.go.dev/github.com/stretchr/testify@v1.9.0/assert#Equal?

That way someone can interrupt the test if they want to